src/core.jl

```diff
@@ -348,9 +348,10 @@ end
         if isnull(get_node_by_name(graph, node_def.name))
             # First try to directly add this node to the graph
             try
-                Operation(node_def)
+                new_op = Operation(node_def)
```

Comment:
`new_op` variable is never used AFAICT, so keeping it is not useful.
As an aside this function is crazy long, it would be nice to break it into several subfunctions.

Reply:
Yep, this should definitely be refactored sometime.